### PR TITLE
Support .NET Framework 4.6 as a base version

### PR DESCRIPTION
**Pull request by pvbchak**
Changes for #164

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@pvbchak__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: src/Microsoft.Azure.Mobile.Server.Storage/StorageDomainManager.cs [hunk 1]
```diff
@@ -752,6 +752,7 @@ internal static ODataQuerySettings GetDefaultQuerySettings()
             };
         }
 
+        [SuppressMessage("Microsoft.Globalization", "CA1305:SpecifyIFormatProvider")]
```
Comment (shrishrirang) [inline src/Microsoft.Azure.Mobile.Server.Storage/StorageDomainManager.cs]: Use invariant culture instead of SuppressMessage

File: src/Microsoft.Azure.Mobile.Server.Authentication/Extensions/IPrincipalExtensions.cs [hunk 2]
```diff
@@ -101,11 +101,9 @@ internal static void PopulateProviderCredentials(TokenEntry tokenEntry, Provider
         {
             if (tokenEntry.UserClaims != null)
             {
-                credentials.Claims = new Dictionary<string, string>();
                 Collection<Claim> userClaims = new Collection<Claim>();
                 foreach (ClaimSlim claim in tokenEntry.UserClaims)
                 {
-                    credentials.Claims[claim.Type] = claim.Value;
```
Comment (shrishrirang) [inline src/Microsoft.Azure.Mobile.Server.Authentication/Extensions/IPrincipalExtensions.cs]: What change is this?

File: src/Microsoft.Azure.Mobile.Server.Authentication/ProviderCredentials.cs [hunk 3]
```diff
@@ -47,15 +47,6 @@ protected ProviderCredentials(string providerName)
...
         /// </summary>
-        /// <remarks>This property is obsolete. Use the UserClaims property instead.</remarks>
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly", Justification = "Setter is required for serialization."), JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        [Obsolete("Use the UserClaims property instead.", error: false)]
-        public IDictionary<string, string> Claims { get; set; }
-
-        /// <summary>
-        /// Gets or sets the collection of additional claims associated with this instance. May
-        /// be null or empty.
-        /// </summary>
```
Comment (shrishrirang) [inline src/Microsoft.Azure.Mobile.Server.Authentication/ProviderCredentials.cs]: Let's add this to the list of changes while doing a release.

File: ServerSdk.proj [hunk 4]
```diff
@@ -52,7 +52,7 @@
   </Target>
   
   <UsingTask TaskName="Xunit.Runner.MSBuild.xunit"
-             AssemblyFile="packages\xunit.runner.msbuild.2.0.0\build\portable-net45+win+wpa81+wp80+monotouch+monoandroid+Xamarin.iOS\xunit.runner.msbuild.dll" />
+             AssemblyFile="packages\xunit.runner.msbuild.2.1.0\build\portable-net45+win8+wp8+wpa81\xunit.runner.msbuild.dll" />
```
Comment (shrishrirang) [inline ServerSdk.proj]: Any change of behavior for mono* ?

File: src/Microsoft.Azure.Mobile.Server.Swagger/MobileAppAuthenticationFilter.cs [hunk 5]
```diff
@@ -42,14 +42,16 @@ public void Apply(Operation operation, SchemaRegistry schemaRegistry, ApiDescrip
             if (authorized.Any())
             {
                 if (operation.security == null)
+                {
                     operation.security = new List<IDictionary<string, IEnumerable<string>>>();
+                }
 
-                var oAuthRequirements = new Dictionary<string, IEnumerable<string>>
+                var OAuthRequirements = new Dictionary<string, IEnumerable<string>>
                 {
```
Comment (shrishrirang) [inline src/Microsoft.Azure.Mobile.Server.Swagger/MobileAppAuthenticationFilter.cs]: curious to know what is the styleCop rule we are violating..

File: src/Microsoft.Azure.Mobile.Server.Login/AppServiceLoginHandler.cs [hunk 6]
```diff
@@ -67,7 +67,7 @@ internal static JwtSecurityToken CreateTokenFromClaims(IEnumerable<Claim> claims
                 Subject = new ClaimsIdentity(claims),
             };
 
-            JwtSecurityTokenHandler securityTokenHandler = new JwtSecurityTokenHandler();
+            var securityTokenHandler = new JwtSecurityTokenHandler() { SetDefaultTimesOnTokenCreation = false };            
             return securityTokenHandler.CreateToken(tokenDescriptor) as JwtSecurityToken;
```
Comment (shrishrirang) [inline src/Microsoft.Azure.Mobile.Server.Login/AppServiceLoginHandler.cs]: So the default behavior of the JwtSecurityHandler has changed not to set the creation time and we are making this change to maintain previous behavior? Any reason why we don't want to go with the new behavior? I am not familiar with the code.. We can have Brett look at this change.

File: src/Microsoft.Azure.Mobile.Server.Storage/StorageDomainManager.cs (hunk 1 above)
Comment (jvano) [inline src/Microsoft.Azure.Mobile.Server.Storage/StorageDomainManager.cs]: Yes, just use the CultureInfo.InvariantCulture in the AppendFormat method

---
In reply to: [83925475](https://github.com/Azure/azure-mobile-apps-net-server/pull/166#discussion_r83925475) [](ancestors = 83925475)

File: src/Microsoft.Azure.Mobile.Server.Authentication/ProviderCredentials.cs (hunk 3 above)
Comment (pvbchak) [inline src/Microsoft.Azure.Mobile.Server.Authentication/ProviderCredentials.cs]: Will Send another PR when we have final build number.

File: src/Microsoft.Azure.Mobile.Server.Storage/StorageDomainManager.cs (hunk 1 above)
Comment (pvbchak) [inline src/Microsoft.Azure.Mobile.Server.Storage/StorageDomainManager.cs]: Fixed.

File: src/Microsoft.Azure.Mobile.Server.Authentication/Extensions/IPrincipalExtensions.cs (hunk 2 above)
Comment (pvbchak) [inline src/Microsoft.Azure.Mobile.Server.Authentication/Extensions/IPrincipalExtensions.cs]: Claims is marked as obselete (use UserClaims instead) prior to this. Since we are doing a major version release, decided to remove it completely.

File: src/Microsoft.Azure.Mobile.Server.Login/AppServiceLoginHandler.cs (hunk 6 above)
Comment (brettsam) [inline src/Microsoft.Azure.Mobile.Server.Login/AppServiceLoginHandler.cs]: This is a separate package for people wanting to do CustomAuth and issue tokens that are consumable by EasyAuth. The important thing is that the JWTs issued here are usable by EasyAuth. Since I'm guessing EasyAuth hasn't changed, I'd lean towards making the behavior match previous releases, even if that means making a code change like this.

**Comment by shrishrirang** [comment]
LGTM
